tortoise_interpreter.py

```diff
@@ -20,41 +19,59 @@ def sanitize_program(program):
 
 
 def navigate_command_functions(COMMANDS, program):
-    # Comment
-    return [f'{i}. {COMMANDS[command[0]](command)}' for i, command in enumerate(program, 1)]
+    command_to_parse = []
+    pen_state = ''
+    for i, command in enumerate(program, 1):
+        new_pen_state = get_pen_state(command, pen_state)
+        command_to_parse.append(f'{i}. {COMMANDS[command[0]](command, pen_state, new_pen_state)}')
+        pen_state = new_pen_state
+    return command_to_parse
+
 
+def pen_up(command, pen_state, new_pen_state):
+    if new_pen_state != pen_state:
+        return f'PEN UP'
+    return f'PEN UP (ignored)'
```

Comment:
Is there a simpler way to write this? One that doesn't require the "PEN UP" string to be there twice?